### PR TITLE
56 pull chicago api ridership data

### PR DESCRIPTION
**Pull request by JPMartinezClaeys**
The script has the functionalities to extract daily data from the Chicago Portal for subway and buses.
I additionally created functions for ingesting data into the database. Since the models are not yet pushed from #27 those are not supposed to run, but I pushed it as an example code for those ingesting data.

If I left it uncommented I failed the tests, but here is a screenshot of the tests (which I'm expecting to add a few more) passing.

Screen Shot 2024-05-02 at 8 57 52 PM

### CODE REVIEW

**Review by mbjackson-capp — APPROVED** [review]

File: tests/test_extract_chicago_ridership.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            143,
+        ),
+        (
+            DATASETS["SUBWAY_RIDERSHIP"]["URL"],
+            datetime.datetime(2023, 3, 4, tzinfo=CHI_TZ),
+            143,
+        ),
+    ],
+)
+def test_extract_daily_data(url, date, expected):
```
Comment (jimenasalinas) [inline tests/test_extract_chicago_ridership.py:35]: I've taken a look at the data extraction and ingestion scripts and they generally look good. For the tests section, should we add a test to check that we're not adding duplicate values into the db?

**Review by jimenasalinas — APPROVED** [review]
The code looks good to me! I just added a comment about maybe adding an extra test to ensure we don't add duplicate values to the db at the route_id/station_id + date level. But maybe the db settings already account for this?

**Comment by JPMartinezClaeys** [comment]
Asking for a re-review here @mbjackson-capp @jimenasalinas @meganhmoore .
The code has been refactored and it's ready to push station data (Some station ridership data has been ingested). I'm facing a problem with Route data but I'll open a new issue to not delay merging this branch.

**Review by jimenasalinas — APPROVED** [review]

File: app/scripts/extract_chi_ridership_data.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            )
+            obs.save()
+        except:
+            print(
+                f"Ingestion for row {row['station_id']} - {row['date']} not succesful"
+            )
+
+
+def run():
+    start = datetime.datetime(2023, 1, 1, tzinfo=CHI_TZ)
```
Comment (meganhmoore) [inline app/scripts/extract_chi_ridership_data.py:227]: same TODO on tz as NYC

File: app/scripts/extract_chi_ridership_data.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    end = datetime.datetime(2023, 2, 1, tzinfo=CHI_TZ)
+    ingest_bus_ridership(start_date=start, end_date=end)
+    # ingest_subway_ridership(start_date=start, end_date=end)
+
+
+# if __name__ == "__main__":
```
Comment (meganhmoore) [inline app/scripts/extract_chi_ridership_data.py:233]: delete if. no longer needed

**Review by meganhmoore — APPROVED** [review]
